### ISSUE TITLE
Add new Service Invocation metrics to proto-based Dapr gRPC proxy

### ISSUE THREAD

**Issue by antontroshin** [dapr/dapr#6459]
## In what area(s)?

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release

## Describe the feature

Add new Service Invocation metrics to proto-based Dapr's gRPC proxy.
The metrics were added in this issue https://github.com/dapr/dapr/issues/5484

Currently, the metrics are missing, to replicate, follow the guide: https://docs.dapr.io/developing-applications/building-blocks/service-invocation/howto-invoke-services-grpc/
After running the server or client as instructed and making sure requests went through, access the metrics port logged in `metrics server started on: <port number>` line, the metrics won't be there.

The only usage found in `direct_messaging.go` https://github.com/dapr/dapr/blob/0c8292c590208bd38886245de7ee1b66b84e4779/pkg/messaging/direct_messaging.go#L274
https://github.com/dapr/dapr/blob/0c8292c590208bd38886245de7ee1b66b84e4779/pkg/messaging/direct_messaging.go#L290

## Release Note

RELEASE NOTE: **ADD** Service Invocation metrics when using proto-based Dapr's gRPC proxying

**Comment by artursouza** [comment]
This should also be validated in E2E tests to make sure it is working and no regression is added in the future.

**Comment by AnnuCode** [comment]
Hi, I tried adding service invocation metrics in `pkg/messaging/grpc_proxy.go` in the `intercept` method, but it didn't work. The metrics do not appear on the metrics server address. I also tried adding the invocation metrics in `pkg/grpc/proxy/handler.go` file in the `handler` method, and it still didn't work. Am I missing some fundamental concept in the way proxying of grpc requests is happening? I thought the `grpc_proxy.go` file of `messaging` package is where the proxying is being executed. Any suggestions would be helpful, thank you.
